tests/integration/defs/stress_test/stress_test.py

```diff
@@ -0,0 +1,941 @@
+                             tp_size=config.tp_size,
+                             memory_requirement=config.memory_requirement,
+                             backend=backend_param)
+
+    # Call the existing stress_test function with the new config and test mode
+    stress_test(new_config, test_mode)
+
+
+def stress_test(config, test_mode):
+    """Test LLM model performance using trtllm-serve and genai-perf.
+
+    This function supports multiple testing modes controlled by the --test-mode option:
+    - "stress-test": Runs the measure capacity stage first, then the stress stage, using the same server instance.
+    - "stress-stage-alone": Performs only the stress stage with customized stress_concurrency and calculated request count.
+
+    Args:
+        config: Model configuration for the test (injected by pytest.mark.parametrize)
+        test_mode: Test mode from the --test-mode option ("stress-test" or "stress-stage-alone")
+    """
+    # Test mode handling - determine which tests to run
+    if test_mode == "stress-test":
+        run_performance = True
+        run_stress = True
+    elif test_mode == "stress-stage-alone":
+        run_performance = False
+        run_stress = True
+    else:
+        pytest.skip(
+            f"Skipping test for unsupported mode: {test_mode}. Supported modes: stress-test, stress-stage-alone"
+        )
+        return  # Exit early
+
+    # Skip if not enough GPU memory
+    if get_device_memory() < config.memory_requirement:
+        pytest.skip(
+            f"Not enough GPU memory. Required: {config.memory_requirement}GB")
+
+    # Skip if not enough GPUs for tensor parallelism
+    if get_device_count() < config.tp_size:
+        pytest.skip(f"Not enough GPUs. Required: {config.tp_size}")
+
+    # Get full model path
+    model_path = get_model_path(config.model_dir)
+    model_name = config.model_name
+
+    # Initialize server config that will be used for all tests
+    test_server_config = ServerConfig()
+
+    # Define test configurations
+    performance_config = PerformanceParams() if run_performance else None
+    stress_config = StressTestConfig(
+        model_config=config,
+        server_config=test_server_config) if run_stress else None
+
+    # Check if server is already running
+    is_healthy, _ = check_server_health(test_server_config.url,
+                                        test_server_config.health_check_timeout)
+    if is_healthy:
+        raise RuntimeError(
+            f"Server is already running at {test_server_config.url}. Please stop it manually before running the stress test."
+        )
+
+    # Start server
+    print_info("Starting trtllm-serve server...")
+    print_info(f"Model path: {model_path}")
+
+    # Verify that model path exists
+    if not os.path.exists(model_path):
+        raise RuntimeError(f"Model path does not exist: {model_path}")
+
+    # For tp_size > 1, use multiple GPUs
+    gpu_list = ",".join(str(i) for i in range(config.tp_size))
```

Comment:
I suggest to not set this inside the test case.
And let outside world to set it.

Setting it to 0~n-1 here equals to not set it actually.

Sometimes QA or other dev want to set GPUs if their server has multiple different GPUs, like both A100 and H100.

Reply:
Agree, let me figure out this